doc/source/guidelines/optional_packages.rst

```diff
@@ -0,0 +1,248 @@
+.. _optional_packages:
+
+Handling Optional Packages
+==========================
+
+The ``setuptools``, ``flit``, and ``poetry`` build systems all allow you to
+declare dependencies that only get installed under specific circumstances. This
+makes it convenient to declare dependencies for ancillary functions such as
+"plotting", "tests", or "docs". This allows you to pragmatically integrate
+optional dependencies which can be installed as optional requirements rather
+than individual packages.
+
+You may want to have optional packages for your pyansys library for a variety
+of reasons, including:
+
+- **Not all users will want to use the feature.** - For example, you might want
+  to make the usage of `matplotlib <https://matplotlib.org/`_ or `pyvista
+  <https://docs.pyvista.org/>`_ if you expect your pyansys library to be used
+  primarily for headless scripting rather than visualization.
```

Comment:
```suggestion
- **Not all users will want to use the feature.** - For example, you might want
  to make the usage of `matplotlib <https://matplotlib.org/`_ or `pyvista
  <https://docs.pyvista.org/>`_ optional if you expect your PyAnsys library to be used
  primarily for headless scripting rather than visualization.
```